src/bicep-registry/README.md

```diff
@@ -0,0 +1,229 @@
+
+1. Modules cannot target multiple scopes.
+2. Scope functions behave differently for each targeted scope.
+3. No mechanism for conditional logic based on scope.
+
+Our templating language supports "scope directives" that enable you to:
+
+1. Leverage existing Bicep developer tooling for the primary scope.
+2. Target multiple scopes.
+3. Conditionally include a single line.
+4. Conditionally include a multiple lines.
+
+### Scope directives
+
+Scope directives are single-line comments that reference one or more Bicep-supported scopes. Scopes are prefixed with `@` and are separated by spaces. Do not include additional characters after the scope directive.
+
+The following example references only the resource group scope:
+
+```bicep
+// @resourceGroup
+```
+
+The following example references all scopes (note order is not important):
+
+```bicep
+// @resourceGroup @subscription @managementGroup @tenant
+```
+
+Refer to the sections below for how to use scope directives.
+
+### Targeting multiple scopes
+
+Modules are designed to work with a single scope. To target multiple scopes, include a scope directive at the top of the file followed by an empty line. Both lines are removed during the build process.
+
+The following example tells the build script to generate modules targeted at resource groups, subscriptions, and management groups:
+
+```bicep
+// @resourceGroup @subscription @managementGroup
+
+@description('The name of the object.')
+param name string
+```
+
+### Conditional lines
+
+Conditional lines are used to include (or exclude) a single line within a Bicep module. To use a conditional line, add a scope directive to the end of the desired line.
+
+The following example includes 2 conditional lines to represent allowed values that are only supported for specific scopes. `value1` is allowed for resource groups and subscriptions, and `value3` is allowed for tenant deployments:
+
+```bicep
+@allowed([
+  'value1' // @resourceGroup @subscription
+  'value2'
+  //'value3' // @tenant
+])
+```
+
+Note this example is designed to use a primary scope of either resource group or subscription (based on those values being uncommented). Since `value3` is only allowed for tenant deployments, it is commented out by default. This approach ensures the module can be run as-is with existing tooling without modification.
+
+### Conditional blocks
+
+Conditional blocks are used to include (or exclude) multiple lines within a Bicep module. To use a conditional block, add a scope directive immediately before the code block. Conditional blocks end when either another conditional block is specified or an empty line.
+
+The following example includes a single conditional block that is only includes `property1` and `property2` for subscriptions:
+
+```bicep
+resource myResource 'Microsoft.MyProvider/resource' = {
+  name: 'myResource'
+  properties: {
+    // @subscription
+    //// Comments must have 4 slashes
```

Comment:
Ah, but on ln186 the text says that ln182 is commented out which conflicts with this line saying Comments need 4 slashes. Consider defining comments as 4 slashes earlier and more explicitly.

Reply:
The problem on this line is I couldn't think of a way to differentiate a true comment (like this line) and a commented out conditional block. I agree `////` feels odd. I thought about other characters for comments, but didn't think that made sense with indented comments (see line 219). This felt more consistent and didn't require custom knowledge.
```bicep
// @subscription
//// Standard comment
// @resourceGroup 
//   // Indented comment

// @subscription
//! Non-standard character
// @resourceGroup 
//   ! Indented comment
```
The good news is whatever we choose is internal to us. We're not building a public API. If we were, I'd prefer to evaluate the line to determine if it's a comment or not.